tests/components/tcp/test_binary_sensor.py

```diff
@@ -1,62 +1,57 @@
-
-    def test_setup_platform_invalid_config(self):
-        """Check the invalid configuration."""
-        with assert_setup_component(0):
-            assert setup_component(
-                self.hass,
-                "binary_sensor",
-                {"binary_sensor": {"platform": "tcp", "porrt": 1234}},
-            )
-
-    @patch("homeassistant.components.tcp.sensor.TcpSensor.update")
-    def test_setup_platform_devices(self, mock_update):
-        """Check the supplied config and call add_entities with sensor."""
-        add_entities = Mock()
-        ret = bin_tcp.setup_platform(None, test_tcp.TEST_CONFIG, add_entities)
-        assert ret is None
-        assert add_entities.called
-        assert isinstance(add_entities.call_args[0][0][0], bin_tcp.TcpBinarySensor)
-
-    @patch("homeassistant.components.tcp.sensor.TcpSensor.update")
-    def test_is_on_true(self, mock_update):
-        """Check the return that _state is value_on."""
-        sensor = bin_tcp.TcpBinarySensor(self.hass, test_tcp.TEST_CONFIG["sensor"])
-        sensor._state = test_tcp.TEST_CONFIG["sensor"][tcp.CONF_VALUE_ON]
-        print(sensor._state)
-        assert sensor.is_on
-
-    @patch("homeassistant.components.tcp.sensor.TcpSensor.update")
-    def test_is_on_false(self, mock_update):
-        """Check the return that _state is not the same as value_on."""
-        sensor = bin_tcp.TcpBinarySensor(self.hass, test_tcp.TEST_CONFIG["sensor"])
-        sensor._state = "{} abc".format(
-            test_tcp.TEST_CONFIG["sensor"][tcp.CONF_VALUE_ON]
+@pytest.fixture
+def mock_update():
+    """Pytest fixture for tcp sensor update."""
+    with patch("homeassistant.components.tcp.sensor.TcpSensor.update") as mock_update:
+        yield mock_update.return_value
+
+
+async def test_setup_platform_valid_config(hass):
+    """Check a valid configuration."""
+    with assert_setup_component(0, "binary_sensor"):
+        assert await async_setup_component(hass, "binary_sensor", test_tcp.TEST_CONFIG)
+
+
+async def test_setup_platform_invalid_config(hass):
+    """Check the invalid configuration."""
+    with assert_setup_component(0):
+        assert await async_setup_component(
+            hass,
+            "binary_sensor",
+            {"binary_sensor": {"platform": "tcp", "porrt": 1234}},
         )
-        assert not sensor.is_on
+
+
+def test_setup_platform_devices(mock_update):
+    """Check the supplied config and call add_entities with sensor."""
+    add_entities = Mock()
+    ret = bin_tcp.setup_platform(None, test_tcp.TEST_CONFIG, add_entities)
+    assert ret is None
+    assert add_entities.called
+    assert isinstance(add_entities.call_args[0][0][0], bin_tcp.TcpBinarySensor)
+
+
+def test_is_on_true(hass, mock_update):
+    """Check the return that _state is value_on."""
+    sensor = bin_tcp.TcpBinarySensor(hass, test_tcp.TEST_CONFIG["sensor"])
+    sensor._state = test_tcp.TEST_CONFIG["sensor"][tcp.CONF_VALUE_ON]
+    print(sensor._state)
```

Comment:
Remember to remove this when troubleshooting is done before merge.

Reply:
done